src/odemis/acq/fastem.py

```diff
@@ -642,16 +646,24 @@ def acquire_roa(self, dataflow):
 
             self.move_stage_to_next_tile()  # move stage to next field image position
             self._scanner.blanker.value = False  # unblank the beam
-            # TODO remove the below temporary fix when a proper solution is found
-            # The temporary fix has been applied so that the ROAs acquisition can still continue upon failure
-            # of correcting the beam shift
-            try:
-                self.correct_beam_shift()  # correct the shift of the beams caused by the parasitic magnetic field.
-            except Exception:
-                logging.exception("Correcting the beam shift failed, check if the image quality is still good.")
-                # In case of failure save the ccd image
-                ccd_image = self._ccd.data.get(asap=False)
-                fastem_util.save_image(self.path, f"{self.field_idx}_after.tiff", ccd_image)
+
+            prev_beam_shift = self._beamshift.shift.value
+            if field_idx in beam_shift_indices or beam_shift_failed:
+                logging.debug(f"Will run beam shift correction for field index {field_idx}")
+                try:
+                    new_beam_shift = self.correct_beam_shift()
+                    # The difference in x or y should not be larger than 2 micrometers
+                    if any(map(lambda x, y: abs(x - y) > 2e-6, new_beam_shift, prev_beam_shift)):
+                        raise ValueError(
+                            "Difference in beam shift is larger than 2 µm, therefore it most likely failed."
```

Comment:
Can you add the previous and new beam shift values to the log, so that it's easier to see how bad it went (or maybe there's something fishy in the values).